src/arch/x86/kernel/smp/smp.c

```diff
@@ -116,10 +116,12 @@ void smp_send_resched(int cpu_id) {
 
 void resched(void) {
 	extern void sched_post_switch(void);
+	extern void sched_post_switch_noyield(void);
 
 	lapic_send_eoi();
 
-	sched_post_switch();
+	//sched_post_switch();
+	sched_post_switch_noyield();
```

Comment:
Maybe bow use SMP macro?
Or you are sure that it is for all type of systems

Reply:
I think it's no need to do. Because this function is only called in [sched_tick](https://github.com/embox/embox/blob/a6d2f9b713dc40f89c2a5a59e15dacd03fb96f7a/src/kernel/sched/sched_ticker.c#L31), which already used SMP macro .